@griffel/react@1.6.2 breaking change: src/*.d.ts switched to default React import, breaking consumers without esModuleInterop

## Bug Report

### Version
`@griffel/react@1.6.2` (published 2026-04-29)

### Description

`@griffel/react@1.6.2` introduced a **breaking change in a minor version** in its TypeScript declaration files. Both `src/RendererContext.d.ts` and `src/TextDirectionContext.d.ts` changed the React import from:

```ts
// 1.5.x — works without esModuleInterop
import * as React from 'react';
```

to:

```ts
// 1.6.2 — requires esModuleInterop: true
import React from 'react';
```

Because `@types/react` uses `export = React` (CommonJS-style), a default import requires `esModuleInterop: true`. Many consumers — particularly those targeting strict or minimal TypeScript configs — do not enable this flag.

### Impact

Any project that:
- depends on `@griffel/react` with a range that resolves to `1.6.x` (e.g. `^1.5.x`)
- does **not** set `esModuleInterop: true` in their tsconfig

will get a hard TypeScript error:

```
error TS1259: Module '@types/react/index' can only be default-imported using the 'esModuleInterop' flag
```

This broke `microsoft/fluentui`'s minimum-TypeScript-bar integration test today. See the workaround PR: https://github.com/microsoft/fluentui/pull/36075

### Expected behavior

A minor version bump should not require consumers to add `esModuleInterop`. The import style should revert to `import * as React from 'react'`, or this change should be released as a major version.

### Suggested fix

Revert the import style in `src/RendererContext.d.ts` and `src/TextDirectionContext.d.ts` to `import * as React from 'react'`, and publish a `1.6.3` patch.

--- a/change/@griffel-react-1000e3be-4b8a-4ff6-9741-ac721219a25e.json
+++ b/change/@griffel-react-1000e3be-4b8a-4ff6-9741-ac721219a25e.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix(react): emit interop-free types so consumers without esModuleInterop can compile",
+  "packageName": "@griffel/react",
+  "email": "olfedias@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/react/src/RendererContext.tsx
+++ b/packages/react/src/RendererContext.tsx
@@ -2,7 +2,7 @@
 
 import { createDOMRenderer, rehydrateRendererCache } from '@griffel/core';
 import type { GriffelRenderer } from '@griffel/core';
-import React from 'react';
+import { createContext, useContext, useMemo, type FC, type ReactNode } from 'react';
 
 import { canUseDOM } from './utils/canUseDOM.js';
 
@@ -18,19 +18,19 @@ export interface RendererProviderProps {
   /**
    * Content wrapped by the RendererProvider
    */
-  children: React.ReactNode;
+  children: ReactNode;
 }
 
 /**
  * @private
  */
-const RendererContext = /*#__PURE__*/ React.createContext<GriffelRenderer>(/*#__PURE__*/ createDOMRenderer());
+const RendererContext = /*#__PURE__*/ createContext<GriffelRenderer>(/*#__PURE__*/ createDOMRenderer());
 
 /**
  * @public
  */
-export const RendererProvider: React.FC<RendererProviderProps> = ({ children, renderer, targetDocument }) => {
-  React.useMemo(() => {
+export const RendererProvider: FC<RendererProviderProps> = ({ children, renderer, targetDocument }) => {
+  useMemo(() => {
     if (canUseDOM()) {
       // "rehydrateCache()" can't be called in effects as it needs to be called before any component will be rendered to
       // avoid double insertion of classes
@@ -47,5 +47,5 @@ export const RendererProvider: React.FC<RendererProviderProps> = ({ children, re
  * @private Exported as "useRenderer_unstable" use it on own risk. Can be changed or removed without a notice.
  */
 export function useRenderer(): GriffelRenderer {
-  return React.useContext(RendererContext);
+  return useContext(RendererContext);
 }

--- a/packages/react/src/TextDirectionContext.tsx
+++ b/packages/react/src/TextDirectionContext.tsx
@@ -1,6 +1,6 @@
 'use client';
 
-import React from 'react';
+import { createContext, useContext, type FC, type ReactNode } from 'react';
 
 export interface TextDirectionProviderProps {
   /** Indicates the directionality of the element's text. */
@@ -9,18 +9,18 @@ export interface TextDirectionProviderProps {
   /**
    * Content wrapped by the TextDirectionProvider.
    */
-  children: React.ReactNode;
+  children: ReactNode;
 }
 
 /**
  * @private
  */
-const TextDirectionContext = /*#__PURE__*/ React.createContext<'ltr' | 'rtl'>('ltr');
+const TextDirectionContext = /*#__PURE__*/ createContext<'ltr' | 'rtl'>('ltr');
 
 /**
  * @public
  */
-export const TextDirectionProvider: React.FC<TextDirectionProviderProps> = ({ children, dir }) => {
+export const TextDirectionProvider: FC<TextDirectionProviderProps> = ({ children, dir }) => {
   return <TextDirectionContext.Provider value={dir}>{children}</TextDirectionContext.Provider>;
 };
 
@@ -30,5 +30,5 @@ export const TextDirectionProvider: React.FC<TextDirectionProviderProps> = ({ ch
  * @private
  */
 export function useTextDirection() {
-  return React.useContext(TextDirectionContext);
+  return useContext(TextDirectionContext);
 }

--- a/packages/react/src/useInsertionEffect.ts
+++ b/packages/react/src/useInsertionEffect.ts
@@ -1,7 +1,10 @@
 'use client';
 
-import React from 'react';
+import * as React from 'react';
 
+/**
+ * @internal
+ */
 export const useInsertionEffect: typeof React.useInsertionEffect | undefined =
   // @ts-expect-error Hack to make sure that `useInsertionEffect` will not cause bundling issues in older React versions
   // eslint-disable-next-line no-useless-concat

--- a/packages/react/src/useInsertionEffect.ts
+++ b/packages/react/src/useInsertionEffect.ts
@@ -1,12 +1,11 @@
 'use client';
 
-// @ts-ignore — default import is intentional (avoids the webpack namespace helper that `import * as` adds); this file is `@internal` and is stripped from the published `.d.ts`.
 import React from 'react';
 
 /**
  * @internal
  */
 export const useInsertionEffect: typeof React.useInsertionEffect | undefined =
-  // @ts-ignore — Hack to make sure that `useInsertionEffect` will not cause bundling issues in older React versions
+  // @ts-expect-error Hack to make sure that `useInsertionEffect` will not cause bundling issues in older React versions
   // eslint-disable-next-line no-useless-concat
   React['useInsertion' + 'Effect'] ? React['useInsertion' + 'Effect'] : undefined;

--- a/packages/react/src/useInsertionEffect.ts
+++ b/packages/react/src/useInsertionEffect.ts
@@ -1,11 +1,12 @@
 'use client';
 
-import * as React from 'react';
+// @ts-ignore — default import is intentional (avoids the webpack namespace helper that `import * as` adds); this file is `@internal` and is stripped from the published `.d.ts`.
+import React from 'react';
 
 /**
  * @internal
  */
 export const useInsertionEffect: typeof React.useInsertionEffect | undefined =
-  // @ts-expect-error Hack to make sure that `useInsertionEffect` will not cause bundling issues in older React versions
+  // @ts-ignore — Hack to make sure that `useInsertionEffect` will not cause bundling issues in older React versions
   // eslint-disable-next-line no-useless-concat
   React['useInsertion' + 'Effect'] ? React['useInsertion' + 'Effect'] : undefined;

--- a/packages/react/src/utils/canUseDOM.ts
+++ b/packages/react/src/utils/canUseDOM.ts
@@ -1,5 +1,7 @@
 /**
  * Verifies if an application can use DOM.
+ *
+ * @internal
  */
 export function canUseDOM(): boolean {
   return typeof window !== 'undefined' && !!(window.document && window.document.createElement);

--- a/packages/react/src/utils/isInsideComponent.ts
+++ b/packages/react/src/utils/isInsideComponent.ts
@@ -1,6 +1,5 @@
 'use client';
 
-// @ts-ignore — default import is intentional (avoids the webpack namespace helper that `import * as` adds); this file is `@internal` and is stripped from the published `.d.ts`.
 import React from 'react';
 
 function getDispatcher() {

--- a/packages/react/src/utils/isInsideComponent.ts
+++ b/packages/react/src/utils/isInsideComponent.ts
@@ -1,6 +1,6 @@
 'use client';
 
-import React from 'react';
+import * as React from 'react';
 
 function getDispatcher() {
   try {
@@ -17,6 +17,9 @@ function getDispatcher() {
   }
 }
 
+/**
+ * @internal
+ */
 export function isInsideComponent() {
   // React 18 always logs errors if a dispatcher is not present:
   // https://github.com/facebook/react/blob/42f15b324f50d0fd98322c21646ac3013e30344a/packages/react/src/ReactHooks.js#L26-L36

--- a/packages/react/src/utils/isInsideComponent.ts
+++ b/packages/react/src/utils/isInsideComponent.ts
@@ -1,6 +1,7 @@
 'use client';
 
-import * as React from 'react';
+// @ts-ignore — default import is intentional (avoids the webpack namespace helper that `import * as` adds); this file is `@internal` and is stripped from the published `.d.ts`.
+import React from 'react';
 
 function getDispatcher() {
   try {

--- a/packages/react/tsconfig.json
+++ b/packages/react/tsconfig.json
@@ -2,9 +2,7 @@
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     "jsx": "react-jsx",
-    "allowJs": true,
-    "esModuleInterop": true,
-    "allowSyntheticDefaultImports": true
+    "allowJs": true
   },
   "files": [],
   "include": [],

--- a/packages/react/tsconfig.lib.json
+++ b/packages/react/tsconfig.lib.json
@@ -5,6 +5,7 @@
     "moduleResolution": "NodeNext",
     "outDir": "../../dist/out-tsc",
     "declaration": true,
+    "stripInternal": true,
     "types": ["node", "environment"]
   },
   "exclude": [

--- a/packages/react/tsconfig.spec.json
+++ b/packages/react/tsconfig.spec.json
@@ -2,6 +2,7 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "outDir": "../../dist/out-tsc",
+    "allowSyntheticDefaultImports": true,
     "typeRoots": ["../../node_modules", "../../node_modules/@types", "../../typings"],
     "types": ["vitest/importMeta", "vite/client", "node", "vitest", "environment"]
   },

--- a/packages/react/tsconfig.storybook.json
+++ b/packages/react/tsconfig.storybook.json
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "emitDecoratorMetadata": true,
     "moduleResolution": "bundler",
+    "allowSyntheticDefaultImports": true,
     "noEmit": true,
     "outDir": ""
   },

